### ISSUE TITLE
Provide USB ids & HID descriptors

### ISSUE THREAD

**Issue by vianpl** [JohnJohanssonChalmers/hp_wireless_keyboard#2]
Hi,
I'm working on a proper kernel fix for this issue and need some information from this keyboard. Would you mind providing the _lsusb_ output and the _cat /sys/kernel/debug/hid/<interface>/rdesc_ output. The _interface_ name will contain the USB ids of the device, which are output by _lsusb_.  Please don't forget to copy the first line of the HID debug report which consists of a string of hex values.

Thanks!

**Comment by pmarini** [comment]
I don't know if this is a reply to my comment, but let me post the required information here:

```
/sys/kernel/debug/hid/0003:0461:4E90.0003# cat rdesc

05 01 09 02 a1 01 09 01 a1 00 05 09 19 01 29 03 15 00 25 01 95 03 75 01 81 02 95 05 81 01 05 01 16 01 f8 26 ff 07 09 30 09 31 75 0c 95 02 81 06 15 81 25 7f 09 38 75 08 95 01 81 06 c0 c0

  INPUT[INPUT]
    Field(0)
      Physical(GenericDesktop.Pointer)
      Application(GenericDesktop.Mouse)
      Usage(3)
        Button.0001
        Button.0002
        Button.0003
      Logical Minimum(0)
      Logical Maximum(1)
      Report Size(1)
      Report Count(3)
      Report Offset(0)
      Flags( Variable Absolute )
    Field(1)
      Physical(GenericDesktop.Pointer)
      Application(GenericDesktop.Mouse)
      Usage(2)
        GenericDesktop.X
        GenericDesktop.Y
      Logical Minimum(-2047)
      Logical Maximum(2047)
      Report Size(12)
      Report Count(2)
      Report Offset(8)
      Flags( Variable Relative )
    Field(2)
      Physical(GenericDesktop.Pointer)
      Application(GenericDesktop.Mouse)
      Usage(1)
        GenericDesktop.Wheel
      Logical Minimum(-127)
      Logical Maximum(127)
      Report Size(8)
      Report Count(1)
      Report Offset(32)
      Flags( Variable Relative )

Button.0001 ---> Key.LeftBtn
Button.0002 ---> Key.RightBtn
Button.0003 ---> Key.MiddleBtn
GenericDesktop.X ---> Relative.X
GenericDesktop.Y ---> Relative.Y
GenericDesktop.Wheel ---> Relative.Wheel
```

There is also another problem: after graphical login the screen turns black and only a random sequence of the F button makes the Desktop back to life.

In the mean time I have upgraded to Ubuntu 18.10.

**Comment by vianpl** [comment]
Hi pmarini,  thanks for your reply!
That's exactly the information I needed, but you copied the mouse HID reports instead of the keyboard ones. It's my fault I should have mentioned that there are two HID reports in /sys/kernel/debug/hid/ that match the USB IDs. So would you mind posting the other one.

Also would you mind telling me the keyboard model name.

**Comment by pmarini** [comment]
I have three folders in /sys/kernel/debug/hid/
To make sure I include the information you need I have tared them:

[hid.zip](https://github.com/JohnJohanssonChalmers/hp_wireless_keyboard/files/2925843/hid.zip)

I'll be able to tell you the keyboard model this evening.

**Comment by vianpl** [comment]
I'm sorry but I think something went wrong. The actual _rdesc_ files are empty :(.

**Comment by pmarini** [comment]
You are right, sorry!
Please check the attached zip
[rdesc.zip](https://github.com/JohnJohanssonChalmers/hp_wireless_keyboard/files/2925926/rdesc.zip)

**Comment by vianpl** [comment]
That's perfect, I added your keyboard to the list of failing ones. Please remember to send me the keyboard model.

If curious here's the kernel mailing list conversation: https://lkml.org/lkml/2019/2/28/510

**Comment by pmarini** [comment]
Thanks for the link and for your support! Just for information, how long will it take to get fixed?

The model is: https://store.hp.com/us/en/pdp/hp-pavilion-wireless-keyboard-and-mouse-800

**Comment by vianpl** [comment]
Well, hopefully I can get it in for kernel version 5.1. Then it really depends on which distribution you're using. I'd give it at the very least 1-2 months :).

**Comment by vianpl** [comment]
FYI I got the fix accepted in mainline: https://lkml.org/lkml/2019/3/27/350 It'll be available in starting from version 5.2. Thanks for the help!

**Comment by pmarini** [comment]
Good news, thanks for the work and update!
